### DISCUSSION TITLE
Can someone provide an example of how to set up segment analytics that works in the `getServerSideProps`?

### DISCUSSION THREAD

**Discussion by zeckdude** [vercel/next.js#43988] [Help]
I've gotten the segment analytics to work fine on the page using `window.analytics.track` but there are times when I want to track an event from inside the `getServerSideProps` and `window` is not available there since the segment analytics is only being imported via a script being added in the document file. Can someone provide an example of how to set up segment analytics that works in the `getServerSideProps`?

**Comment by icyJoseph** [comment]
What is `segment analytics`?

What kind of event are you trying to track from `GSSP`? 

Assuming this is what you mean by [`segment analytics`](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/), then there's a node package you can use within `GSSP`.

https://github.com/segmentio/analytics-next/tree/master/packages/node
